lib/Conch/Route/Device.pm

```diff
@@ -14,128 +14,496 @@ Conch::Route::Device
+
+=item * Request Schema: L<DevicePhase|https://conch.joyent.us/schema/request/DevicePhase>
+
+=back
+
+=cut
+
         # POST /device/:device_id/phase
         $with_device->post('/phase')->to('device#set_phase');
 
         {
             my $with_device_location = $with_device->any('/location');
             $with_device_location->to({ controller => 'device_location' });
 
+
+=head3 C<GET /device/:device_id/location>
+
+Retrieves location data for the current device. Returns a C<409 CONFLICT> if
+the device is not assigned to a rack.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DeviceLocation|https://conch.joyent.us/schema/response/DeviceLocation>
+
+=back
+
+=cut
             # GET /device/:device_id/location
             $with_device_location->get('/')->to('#get');
-            # POST /device/:device_id/location
+
+=head3 C<POST /device/:device_id/location>
+
+Sets the location for a device, given a valid rack id and rack unit. Upon
+success returns C<303 SEE OTHER> with the URL of the device.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Request Schema: L<DeviceLocationUpdate|https://conch.joyent.us/schema/request/DeviceLocationUpdate>
+
+=back
+
+=cut
+
+           # POST /device/:device_id/location
             $with_device_location->post('/')->to('#set');
-            # DELETE /device/:device_id/location
+
+=head3 C<DELETE /device/:device_id/location>
+
+Deletes the location data for a device, provided it has been assigned to a
+location. Returns a C<409 CONFLICT> if the device is not assigned to a rack.
+Upon success returns a C<204 NO CONTENT>.
+
+=over 4
+
+=item * Requires Authentication
+
+=back
+
+=cut
+
+           # DELETE /device/:device_id/location
             $with_device_location->delete('/')->to('#delete');
         }
 
         {
             my $with_device_settings = $with_device->any('/settings');
             $with_device_settings->to({ controller => 'device_settings' });
 
-            # GET /device/:device_id/settings
+
+=head3 C<GET /device/:device_id/settings>
+
+Get all settings for a device as a JSON object.
+
+=over 4
+
+=item * Requires Authentication
+
+=back
+
+=cut
+
+           # GET /device/:device_id/settings
             $with_device_settings->get('/')->to('#get_all');
-            # POST /device/:device_id/settings
+
+=head3 C<POST /device/:device_id/settings>
+
+Overrides all settings for a device with the given payload. Existing settings
+are deactivated even if they are not being replaced with new ones. Returns
+C<200 OK> upon success.
+
+=over 4
+
+=item * Requires R/W Device Authentication
+
+=back
+
+=cut
+
+           # POST /device/:device_id/settings
             $with_device_settings->post('/')->to('#set_all');
 
             my $with_device_settings_with_key = $with_device_settings->any('/#key');
-            # GET /device/:device_id/settings/#key
+
+=head3 C<GET /device/:device_id/settings/#key>
+
+Get a single setting from a device.
+
```

Comment:
What's the response payload here?

Reply:
Should be documented as the `DeviceSetting` schema now.